### PR TITLE
HM_dao8

### PR DESCRIPTION
**Pull request by grom4er**

### CODE REVIEW

File: src/main/webapp/web.xml [hunk 1]
```diff
@@ -11,4 +11,49 @@
...
         <url-pattern>/index</url-pattern>
     </servlet-mapping>
+
+    <servlet>
+        <servlet-name>addDriverToCar</servlet-name>
+        <servlet-class>taxiservice.controllers.cars.AddDriverToCarController</servlet-class>
+    </servlet>
+    <servlet-mapping>
+        <servlet-name>addDriverToCar</servlet-name>
+        <url-pattern>/cars/add_driver_to_car</url-pattern>
```
Comment (Resci) [inline src/main/webapp/web.xml:21]: Bad naming. Check [checklist](https://mate-academy.github.io/jv-program-fulltime/02_jdbc/checklist/08_servlets_practice_checklist.html)

File: src/main/java/taxiservice/controllers/cars/AddDriverToCarController.java [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+
+    protected void doGet(HttpServletRequest req, HttpServletResponse resp)
+            throws ServletException, IOException {
+        req.getRequestDispatcher("/WEB-INF/cars/add_driver_to_car.jsp").forward(req, resp);
+    }
+
+    protected void doPost(HttpServletRequest req, HttpServletResponse resp)
+            throws ServletException, IOException {
+        String carID = req.getParameter("carID");
+        String driverID = req.getParameter("driverID");
```
Comment (Resci) [inline src/main/java/taxiservice/controllers/cars/AddDriverToCarController.java:31]: You need this variables?

**Review by Resci — CHANGES_REQUESTED** [review]

**Review by Resci — APPROVED** [review]

File: src/main/java/taxiservice/controllers/cars/CreateCarController.java [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+    protected void doGet(HttpServletRequest req, HttpServletResponse resp)
+            throws ServletException, IOException {
+        req.getRequestDispatcher("/WEB-INF/cars/create.jsp").forward(req, resp);
+    }
+
+    @Override
+    protected void doPost(HttpServletRequest req, HttpServletResponse resp)
+            throws IOException {
+        String model = req.getParameter("model");
+        String manufactureId = req.getParameter("manufacture id");
```
Comment (dmytro-hryhoruk) [inline src/main/java/taxiservice/controllers/cars/CreateCarController.java:32]:
```suggestion
        String manufactureId = req.getParameter("manufacture_id");
```
This parameter's name is different from the one in jsp. So it doesn't work. Let's test code before commiting it
image

Comment (dmytro-hryhoruk) [reply src/main/java/taxiservice/controllers/cars/CreateCarController.java:32]: check this everywhere

File: src/main/webapp/WEB-INF/index.jsp [hunk 4]
```diff
@@ -1,9 +1,23 @@
...
+  </form>
+  <form action="${pageContext.request.contextPath}/manufacturer/create_manufacturer">
+    <button type="submit">Create a manufacturer</button>
+  </form>
+  <form action="${pageContext.request.contextPath}/drivers/create_driver">
+    <button type="submit">Create driver</button>
+  </form>
+  <form action="${pageContext.request.contextPath}/cars/add_driver_to_car">
+    <button type="submit">Add driver to car</button>
+  </form>
```
Comment (dmytro-hryhoruk) [inline src/main/webapp/WEB-INF/index.jsp:21]: It doesn't work too

**Review by dmytro-hryhoruk — CHANGES_REQUESTED** [review]

**Review by dmytro-hryhoruk — APPROVED** [review]

**Pull request merged**